react/LocationSearch.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react'
+import { Input, IconSearch, IconClear } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+
+import styles from './LocationSearch.css'
+
+const LocationSearch: React.FC = () => {
+  const [inputValue, setInputValue] = useState<string>('')
+
+  const onClear = () => setInputValue('')
+
+  return (
+    <div className={`${styles.locationSearch} w-100`}>
+      <Input
+        label={
+          <FormattedMessage id="place-components.label.autocompleteAddress" />
+        }
+        prefix={<IconSearch color="#134CD8" />}
+        suffix={
+          inputValue.length && (
+            <span
+              role="button"
+              tabIndex={-1}
```

Comment:
should be `tabIndex={0}`, else we will never be able to tab into this element
```suggestion
              tabIndex={0}
```

Reply:
Do you really think we should be able to tab into it? As this behavior wasn't described in the design document, I just copied the behavior from other autocomplete components such as [VTEX Styleguide](https://styleguide.vtex.com/#/Components/Forms/AutocompleteInput), [Algolia](url) and [Material UI](https://material-ui.com/pt/components/autocomplete/).

Reply:
yes, if we don't make it tabbable we would degrade the experience for keyboard and screen reader users

Reply:
instead of making this tabbable, we could add a keydown listener on the input itself, and handle the `escape` key event as the clear., which is actually one of the specification requirements for a combobox (or autocomplete)

![image](https://user-images.githubusercontent.com/10223856/86955941-bf475180-c12e-11ea-9103-e59e8b069f69.png)

Reply:
there is [this example](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html) of a combobox with a listbox, which I think is the concept for the UI of this autocomplete, if you want to read more about accessibility

Reply:
Didn't managed to make this work :(

I tried to use`onKeyDown` and `onKeyPressed` events, but even preventing the default behavior (as suggested by @marcoskwkm), I still can't capture `<Esc>`. On the other hand I can capture almost anything else.

I suppose the styleguide component is forcing some behavior under the hood, but I couldn't identify it on the[ input source code](https://github.com/vtex/styleguide/blob/master/react/components/Input/index.js).

Any ideas?

Reply:
Solved it! 🙃🙃

The problem: I use a Chrome extension that allows me to use Vim keybindings for navigation and it was capturing some keys before the component... that also explains why the components on w3 weren't behaving properly.